src/entities/Checkpoint.ts

```diff
@@ -5,33 +5,76 @@ import { DividendDistribution } from './DividendDistribution';
 import { ShareholderBalance, ErrorCode } from '../types';
 import { PolymathError } from '../PolymathError';
 
+/**
+ * Represents a checkpoint identifier
+ */
 export interface UniqueIdentifiers {
+  /**
+   * st identifier
+   */
   securityTokenId: string;
+  /**
+   * checkpoint id used to query historical balances
+   */
   index: number;
 }
 
+/**
+ * Check if the provided value is of type [[UniqueIdentifiers]]
+ *
+ * @param identifiers - internal checkpoint representation
+ */
 function isUniqueIdentifiers(identifiers: any): identifiers is UniqueIdentifiers {
   const { securityTokenSymbol, index } = identifiers;
 
   return typeof securityTokenSymbol === 'string' && typeof index === 'number';
 }
 
+/**
+ * Represents a single checkpoint instance
+ */
 export interface Params {
+  /**
+   * information for distributing dividends to holders
+   */
   dividendDistributions: DividendDistribution[];
+  /**
+   * symbol of the security token
+   */
   securityTokenSymbol: string;
+  /**
+   * balances for security token holders
+   */
   shareholderBalances: ShareholderBalance[];
+  /**
+   * total number of st market cap
+   */
   totalSupply: BigNumber;
+  /**
+   * date of checkpoint creation
+   */
   createdAt: Date;
 }
 
+/**
+ * Checkpoint implementation used to manage Security Token checkpoint functionality
```

Comment:
Represents a snapshot of the Security Token's supply and Shareholder balances at a certain point in time